### DISCUSSION TITLE
Multiple Argo installations in same cluster

### DISCUSSION THREAD

**Discussion by rogerwelin** [argoproj/argo-cd#5607] [Q&A]
What are best practices regarding having multiple installations of Argo in the same cluster? I fail to find relevant documentation regarding the subject. In my case I want both Argo installations to deploy cluster-wide (not restricted to a single namespace)

**Comment by jgwest** [comment]
Hi @rogerwelin, I've personally not tried this, and I'd be curious if others have experiences here to share, but AFAICT this should work, with the following caveats/notes/suggestions:
- For each of the Argo CD instances, you'll need to create the ClusterRoleBindings/ClusterRoles/ServiceAccounts yourself to ensure that each has the required permissions. The [standard install RBAC resources](https://github.com/argoproj/argo-cd/tree/master/manifests/cluster-rbac) of the cluster-scoped install assume a single Argo CD instance at cluster scope (and you can use the resources listed there as a template)
- Each Argo CD instance should be contained within a different namespace (eg rather than a single `argocd` namespace, instead use `argocd-team-a` and `argocd-team-b`, for example)
- You'll want to keep the Argo CD instances from stepping on each other's toes, primarily by ensuring that they do not both touch the same namespace (or said another way, there should be no intersection between the set of namespaces managed by any two Argo CD instances).
- Both Argo CD versions should be the same (or as close as possible): at a minimum, the CRDs of each should be the same.

Finally, you may wish to consider that it would be more efficient to use a single Argo CD instance at the cluster scope, and setup [Projects](https://argoproj.github.io/argo-cd/user-guide/projects/) and [Argo CD RBAC](https://argoproj.github.io/argo-cd/operator-manual/user-management/) to ensure that appropriate permission constraints are handled for a multitenant environment, if this is your goal.

**Comment by disposab1e** [comment]
@rogerwelin Did you find a good working solution for you? I want to have an instance of ArgoCD for cluster configuration and different instances for teams working in their set of namespaces on the same cluster.
